include/odp/api/spec/ipsec.h

```diff
@@ -1000,6 +1015,24 @@ typedef struct odp_ipsec_out_opt_t {
 	/** Fragmentation mode */
 	odp_ipsec_frag_mode_t frag_mode;
 
+	/** Union of IP parameters */
+	union {
+		/** Override IPv4 parameters in outer header creation.
+		 *  IP addresses are ignored. */
+		odp_ipsec_ipv4_param_t ipv4;
+
+		/** Override IPv6 parameters in outer header creation.
+		 *  IP addresses are ignored. */
+		odp_ipsec_ipv6_param_t ipv6;
+	};
+
+	/** TFC padding length
+	 *
+	 *  Number of TFC padding bytes added to the packet during IPSEC
+	 *  processing. Implementation guarantees that the padding does not
+	 *  contain any confidential information. */
+	uint32_t tfc_pad_len;
```

Comment:
I have mixed feelings towards this option. Allocating additional buffer space from application-specific pools seems fragile to me. Maybe we can leave this out for now till there is hardware which actually supports generating additional TFC padding?

Reply:
On January 17 pvanleeuwn@insidesecure.com wrote in the lng-odp list:

"However, for the EIP96 - EIP197PP range of solutions our HW /is/ capable of TFC padding insertion, we would just need to expose this through our driver and firmware. So API-wise you might want to support adding n bytes of TFC padding for outbound. "

So apparently such HW exists. And even with pure SW solution, I do not see why buffer space would need to be allocated from application specific pools for the padding data.